compose/ui/ui-test/src/uikitMain/kotlin/androidx/compose/ui/test/ComposeUiTest.uikitMain.test.kt

```diff
@@ -0,0 +1,182 @@
+                    val placeable = measurables.firstOrNull()?.measure(constraints)
+
+                    val width = placeable?.width ?: constraints.minWidth
+                    val height = placeable?.height ?: constraints.minHeight
+
+                    layout(width, height) {
+                        placeable?.placeRelative(0, 0)
+                    }
+                }
+            )
+        }
+        window.rootViewController = composeContainer
+        window.makeKeyAndVisible()
+        waitForExpectation { onDraw }
+    }
+
+    private fun keyboardUserInfo(height: Dp, animated: Boolean): Map<Any?, *> {
+        val keyboardHeight = screenSize.height - height
+        val keyboardDuration = if (animated) 0.25 else 0.0
+        return mapOf(
+            UIKeyboardAnimationDurationUserInfoKey to NSNumber(double = keyboardDuration),
+            UIKeyboardFrameEndUserInfoKey to NSValue.valueWithCGRect(
+                CGRectMake(
+                    x = 0.0,
+                    y = keyboardHeight.value.toDouble(),
+                    width = screenSize.width.value.toDouble(),
+                    height = keyboardHeight.value.toDouble()
+                )
+            ),
+            UIKeyboardAnimationCurveUserInfoKey to NSNumber(unsignedInteger = 0uL)
+        )
+    }
+
+    fun showKeyboard(height: Dp = 216.dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillShowNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(height, animated)
+            )
+        )
+        keyboardHeight = height
+        resizeKeyboard(height, animated)
+    }
+
+    fun resizeKeyboard(newHeight: Dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(newHeight, animated)
+            )
+        )
+        keyboardHeight = newHeight
+    }
+
+    fun hideKeyboard(animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(0.dp, animated)
+            )
+        )
+        keyboardHeight = 0.dp
+    }
+
+    fun <T> waitToEquals(expected: T, actual: () -> T, deadline: Duration = 1.seconds) {
+        waitToEquals({ expected }, actual, deadline)
+    }
+
+    fun <T> waitToEquals(expected: () -> T, actual: T, deadline: Duration = 1.seconds) {
+        waitToEquals(expected, { actual }, deadline)
+    }
+
+    fun <T> waitToEquals(expected: () -> T, actual: () -> T, deadline: Duration = 1.seconds) {
+        waitForExpectation(deadline) {
+            expected() == actual()
+        }
```

Comment:
1. I don't think that we need to redefine asserts here
2. Waiting should be based on `isIdle` check but not on comparing exect/actual from assert

Reply:
As I mentioned, isIdle does not work correctly when we're not bound with manually operating clocks.